File: data/StarCoder2/readable_format_HPEC/cpp/file024.cpp
#include <iostream>
#include <omp.h>

int main() {
    const int len = 100;
    int a[len][len];

    #pragma omp parallel
    {
        #pragma omp single
        {
            #pragma omp taskloop
            for (int i = 0; i < len; ++i) {
                for (int j = 0; j < len; ++j) {
                    a[i][j] = a[i][j] + 1;
                }
            }
        }
    }

    std::cout << "a[50][50] = " << a[50][50] << std::endl;

    return 0;
}

File: data/gpt-4o/readable_format_HPEC/fortran/file024.f90
program DRB095_doall2_taskloop_orig_yes
    use omp_lib
    use DRB095
    implicit none

    integer :: len, i, j
    len = 100
    allocate (a(len,len))

    !$omp parallel
        !$omp single
            !$omp taskloop
            do i = 1, len
                do j = 1, len
                    a(i,j) = a(i,j)+1
                end do
            end do
            !$omp end taskloop
        !$omp end single
    !$omp end parallel

    print 100, a(50,50)
    100 format ('a(50,50) =',i3)

end program
